pkg/provider/aws/resources/cloudwatch.go

```diff
@@ -20,6 +20,27 @@ type (
 	}
 )
 
+type CloudwatchLogGroupCreateParams struct {
+	AppName string
+	Refs    []core.AnnotationKey
+	Name    string
+}
+
+// Create takes in an all necessary parameters to generate the LogGroup name and ensure that the LogGroup is correlated to the constructs which required its creation.
+func (logGroup *LogGroup) Create(dag *core.ResourceGraph, params CloudwatchLogGroupCreateParams) error {
+	logGroup.Name = logGroupSanitizer.Apply(fmt.Sprintf("%s-%s", params.AppName, params.Name))
+	logGroup.ConstructsRef = params.Refs
+
+	existingLogGroup := dag.GetResource(logGroup.Id())
+	if existingLogGroup != nil {
+		graphLogGroup := existingLogGroup.(*LogGroup)
```

Comment:
nit: I think there's a generic form of `GetResource` that will hide this casting.

Reply:
hmmmm i didnt think there was, but ill take another look

Reply:
Every time I've written it, I've ended up removing it before putting up a PR because I ended up not using it. At least the groundwork of renaming the old generic `GetResource` that returned a construct made it in at some point.